packages/date/src/index.js

```diff
@@ -85,7 +90,17 @@ export function setSettings( dateSettings ) {
  *
  * @return {Object} Settings, including locale data.
  */
+export function __experimentalGetSettings() {
+	return settings;
+}
+
+// deprecations
 export function getSettings() {
+	deprecated( 'wp.date.getSettings', {
+		version: '4.2',
+		alternative: 'wp.date.__experimentalGetSettings',
```

Comment:
Given the verbiage logged with the deprecation notice ("Please use ... instead"), we should consider either omitting the alternative or adding a [scary hint](https://github.com/WordPress/gutenberg/pull/9106/files#diff-e373e566d15f8f4196eb0b3998adbdd7R119) to discourage its use.